### DISCUSSION TITLE
Notifications of subscription changes

### DISCUSSION THREAD

**Discussion by joemasilotti** [pay-rails/pay#679] [Q&A]
How can I get notified of subscription changes in Pay?

I see that I can extend the `Pay::Subscription` model with a module, which helps. But I feel like there's got to be a better way than checking `previous_changes` after a commit. Am I missing something?

Here's a gist of the working (but ugly) code and passing tests.

https://gist.github.com/joemasilotti/c0f7f3c4c31f67a81c13d10579af22be

**Comment by excid3** [comment]
Right now, Pay simply syncs the actual data from the APIs, so these don't exist. 

It could be nice to have a feature like this, but it is not really a status as much as it is an event that happened to change the status. I would think it'd be better to emit events on these changes and let Pay users consume them with callbacks instead of trying to save it as a status and then layer callbacks on top.

Also your `resubscribed` status only seems to handle a user resuming a subscription that will cancel in the future. To me, `resubscribed` sounds more like a user who had previously canceled has now created a new subscription. It probably needs a bit more specific naming to differentiate those two. Maybe `resumed` and `resubscribed`? `resumed` would also have to be clear that it's separate from unpausing, but I as long as those have separate names, I think that's clear enough.

**Reply by joemasilotti** [reply]
Emitting events and letting Pay users consume them with callbacks sounds great to me! Are there any examples of this type of workflow in Pay that I can model it after?

`resumed` vs. `resubscribed` sounds good to me.

I'm also assuming there might be some non-Stripe-specific code that I'm missing, especially around pausing/unpausing.
